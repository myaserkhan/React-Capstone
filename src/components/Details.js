import React, { useState, useEffect } from 'react';
// import PropTypes from 'prop-types';
import axios from 'axios';
import { NavLink, useParams } from 'react-router-dom';
import { IconContext } from 'react-icons';
import { BsHouseFill } from 'react-icons/bs';
import './stylesheets/Details.scss';
import pokeball from '../assets/pokeball.png';
// import Capitalize from './utils/Capitalize';

function Details() {
  const [img, setImg] = useState(pokeball);
  const [img1, setImg1] = useState(pokeball);
  const [img2, setImg2] = useState(pokeball);
  const [img3, setImg3] = useState(pokeball);
  const [img4, setImg4] = useState(pokeball);
  const [pokemon, setPokemon] = useState('');
  const params = useParams();
  console.log(pokemon);

  useEffect(() => {
    async function fetchData() {
      const response = await axios.get(`https://pokeapi.co/api/v2/pokemon/${params.id}`);
      setImg(response.data.sprites.other['official-artwork'].front_default);
      setImg1(response.data.sprites.front_default);
      setImg2(response.data.sprites.back_default);
      setImg3(response.data.sprites.front_shiny);
      setImg4(response.data.sprites.back_shiny);
      setPokemon(response.data);
    }
    fetchData();
  }, []);

  return (
    <>
      <div className="detailsContainer">
        <NavLink to="/">
          <IconContext.Provider value={{ className: 'homeBtn' }}>
            <BsHouseFill />
          </IconContext.Provider>
        </NavLink>
        <img src={img} alt="" />

        <div className="photos">
          {/* {!pokemon.length
            ? (<img src={pokemon.sprites.front_default} alt="" />) : (<img src={img} alt="" />)} */}
          <img src={img1} alt="" />
          <img src={img2} alt="" />
          <img src={img3} alt="" />
          <img src={img4} alt="" />
        </div>
      </div>
      <div className="statsContainer">
        <table>
          <tr>
            <td className="property">
              ID:
            </td>
            <td className="info">
              {pokemon.id}
            </td>
          </tr>
          <tr>
            <td className="property">
              Name:
            </td>
            <td className="info">
              {pokemon.name}
            </td>
          </tr>
          <tr>
            <td className="property">
              Base Experience:
            </td>
            <td className="info">
              {pokemon.base_experience}
            </td>
          </tr>
          <tr>
            <td className="property">
              Height:
            </td>
            <td className="info">
              {pokemon.height}
            </td>
          </tr>
          <tr>
            <td className="property">
              Weight:
            </td>
            <td className="info">
              {pokemon.weight}
            </td>
          </tr>
        </table>
      </div>
    </>
  );
}

// Details.propTypes = {
//   id: PropTypes.number.isRequired,
// };

export default Details;
