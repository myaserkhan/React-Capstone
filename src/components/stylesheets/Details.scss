@import '../../App.scss';

.details {
  display: flex;
  flex-direction: column;
  position: relative;
}

.detailsContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: $clr-blue;

  h1 {
    font-size: 1.25rem;
    position: absolute;
    top: 0;
    left: 0;
    color: $clr-white;
    border: 1px solid $clr-white;
    border-radius: 50%;
    padding: 0.25rem;
    margin: 1rem 0 0 1rem;
  }

  h2 {
    color: $clr-yellow-gold;
    font-size: 1.25rem;
    position: absolute;
    margin: 1rem auto;
    top: 0;
    left: 0;
    right: 0;
    text-align: center;
  }

  .homeBtn {
    color: $clr-black-two;
    font-size: 2rem;
    margin: 1rem 1rem 0 0;
    position: absolute;
    top: 0;
    right: 0;
  }

  img {
    margin: 1rem 0;
    width: 375px;
    height: 375px;
  }

  .photos {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    margin: 0 auto;

    img {
      background-color: $clr-black-two;
      border-radius: 1rem;
      margin: 0 1px;
      width: 80px;
      height: 80px;
    }
  }
}

.bottom {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  position: fixed;
  bottom: 0;
  background-color: $clr-black-two;
  width: 100%;
  padding: 0 0 1rem 0;

  .statsContainer {
    display: flex;
    flex-direction: column;
    padding: 1rem;
    list-style-type: none;
    width: 55%;

    .property {
      margin: 0 0 0 1rem;
      color: $clr-white;
      display: flex;
      justify-content: space-between;
    }

    .info {
      color: $clr-yellow-gold;
    }
  }

  .gifPhotos {
    display: flex;
    flex-direction: column;
    margin: 0 4rem 0 0;

    img {
      margin: 0 0 1rem 0;
    }
  }
}

@media screen and (min-width: 768px) {
  .details {
    display: flex;
    flex-direction: column;
    position: relative;

    .detailsContainer {
      display: flex;
      flex-direction: row;
      justify-content: center;
      margin: 0 auto;

      h2 {
        font-size: 2rem;
        position: static;
        margin: 1rem auto;
        bottom: 0;
        left: 0;
        right: 0;
        text-align: center;
      }

      .photos {
        display: flex;
        flex-direction: column;

        img {
          margin: 0.5rem;
        }
      }
    }

    .bottom {
      margin: 0 auto;
      justify-content: center;
      padding: 0 0 2rem 0;

      .statsContainer {
        width: 20%;
        margin: 0 2rem;

        .property {
          justify-content: space-between;
        }
      }
    }

    .gifPhotos {
      margin: 0;
    }
  }
}
