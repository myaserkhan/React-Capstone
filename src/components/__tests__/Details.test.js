import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Details from '../Details';
import store from '../redux/configureStore';

const MockDetails = () => (
  <Provider store={store}>
    <BrowserRouter>
      <Details />
    </BrowserRouter>
  </Provider>
);

describe('Test the Details component', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<MockDetails />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders link', () => {
    render(<MockDetails />);
    const buttonElement = screen.getByRole('button', {
      name: /more details\.\.\./i,
    });
    expect(buttonElement).toBeInTheDocument();
  });
});
