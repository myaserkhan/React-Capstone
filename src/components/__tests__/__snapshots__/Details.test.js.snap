// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Details component renders correctly 1`] = `
<section
  className="details"
>
  <div
    className="detailsContainer"
  >
    <h1 />
    <h2 />
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      <svg
        className="homeBtn"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m8 3.293 6 6V13.5a1.5 1.5 0 0 1-1.5 1.5h-9A1.5 1.5 0 0 1 2 13.5V9.293l6-6zm5-.793V6l-2-2V2.5a.5.5 0 0 1 .5-.5h1a.5.5 0 0 1 .5.5z"
          fillRule="evenodd"
        />
        <path
          d="M7.293 1.5a1 1 0 0 1 1.414 0l6.647 6.646a.5.5 0 0 1-.708.708L8 2.207 1.354 8.854a.5.5 0 1 1-.708-.708L7.293 1.5z"
          fillRule="evenodd"
        />
      </svg>
    </a>
    <img
      alt=""
      src="pokeball.png"
    />
    <div
      className="photos"
    >
      <img
        alt=""
        aria-hidden="true"
        onClick={[Function]}
        src="pokeball.png"
      />
      <img
        alt=""
        aria-hidden="true"
        onClick={[Function]}
        src="pokeball.png"
      />
      <img
        alt=""
        aria-hidden="true"
        onClick={[Function]}
        src="pokeball.png"
      />
    </div>
  </div>
  <div
    className="bottom"
  >
    <ul
      className="statsContainer"
    >
      <li
        className="property"
      >
        Type:
        <span
          className="info"
        />
      </li>
      <li
        className="property"
      >
        Ability 01:
        <span
          className="info"
        />
      </li>
      <li
        className="property"
      >
        Ability 02:
        <span
          className="info"
        />
      </li>
      <li
        className="property"
      >
        Base Exp:
        <span
          className="info"
        />
      </li>
      <li
        className="property"
      >
        Height:
        <span
          className="info"
        />
      </li>
      <li
        className="property"
      >
        Weight:
        <span
          className="info"
        />
      </li>
    </ul>
    <div
      className="gifPhotos"
    >
      <img
        alt=""
        aria-hidden="true"
        onClick={[Function]}
      />
      <button
        onClick={[Function]}
        type="button"
      >
        More Details...
      </button>
    </div>
  </div>
  <footer>
    Copyright ©️ 2022
    <a
      href="https://github.com/myaserkhan"
    >
      myaserkhan
    </a>
  </footer>
</section>
`;
